File: DesktopSample/API/PageAndRDL/ReportWizard/C#/ReportWizard/UI/WizardSteps/BaseStep.cs
using System.ComponentModel;
using System.Windows.Forms;

namespace ActiveReports.Samples.ReportWizard.UI.WizardSteps
{
	public partial class BaseStep : UserControl
	{
		private bool _firstDisplay;
		private string _stepTitle;
		private string _stepDescription;
		private ReportWizardState _wizState;

		public BaseStep()
		{
			_firstDisplay = true;
			_stepTitle = "";
			_stepDescription = "";
			InitializeComponent();
		}

		[Browsable(false)]
		public ReportWizardState ReportWizardState
		{
			get { return _wizState; }
			set { _wizState = value; }
		}

		[Browsable(true)]
		[Description("表示するウィザードのタイトル")]
		[Category("外観")]
		[DefaultValue("")]
		public string Title
		{
			get { return _stepTitle; }
			set { _stepTitle = value; }
		}

		[Browsable(true)]
		[Description("ステップの実装方法を説明するテキスト")]
		[Category("外観")]
		[DefaultValue("")]
		public string Description
		{
			get { return _stepDescription; }
			set { _stepDescription = value; }
		}

		public void OnDisplay()
		{
			OnDisplay(_firstDisplay);
			_firstDisplay = false;
		}

		protected virtual void OnDisplay(bool firstDisplay)
		{
		}

		/// <summary>
		///	ウィザードは前または次のステップへ移動する前に呼び出します。
		/// </summary>
		public virtual void UpdateState()
		{
		}
	}
}

File: DesktopSample/API/PageAndRDL/ReportWizard/VB.NET/ReportWizard/UI/WizardSteps/BaseStep.vb
Imports System.ComponentModel
Imports System.Windows.Forms
Partial Public Class BaseStep
    Inherits UserControl
    Private firstDisplay As Boolean
    Private stepTitle As String
    Private stepDescription As String
    Public Sub New()
        firstDisplay = True
        stepTitle = ""
        stepDescription = ""

        InitializeComponent()
    End Sub
    Private wizState As ReportWizardState
    <Browsable(False)>
    Public Property ReportWizardState() As ReportWizardState
        Get
            Return wizState
        End Get
        Set(ByVal value As ReportWizardState)
            wizState = value
        End Set
    End Property

    <Browsable(True)>
    <Description("表示するウィザードのタイトル")>
    <Category("外観")>
    <DefaultValue("")>
    Public Property Title() As String

        Get
            Return stepTitle
        End Get
        Set(ByVal value As String)
            stepTitle = value
        End Set
    End Property

    <Browsable(True)>
    <Description("ステップの実装方法を説明するテキスト")>
    <Category("外観")>
    <DefaultValue("")>
    Public Property Description() As String
        Get
            Return stepDescription
        End Get
        Set(ByVal value As String)
            stepDescription = value
        End Set
    End Property
    Public Sub OnDisplay()
        OnDisplay(Me.firstDisplay)
        firstDisplay = False
    End Sub
    Protected Overridable Sub OnDisplay(ByVal firstDisplay As Boolean)
    End Sub
    'ウィザードは前または次のステップへ移動する前に呼び出します。
    Public Overridable Sub UpdateState()
    End Sub
End Class
